app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/PreValidateTraderController.scala

```diff
@@ -31,17 +34,48 @@ class PreValidateTraderController @Inject() (
   authAction: AuthAction,
   parseJsonAction: ParseJsonAction,
   preValidateTraderService: PreValidateTraderService,
-  cc: ControllerComponents
+  cc: ControllerComponents,
+  appConfig: AppConfig
 )(implicit ec: ExecutionContext)
     extends BackendController(cc) {
 
-  def submit: Action[JsValue] =
-    (authAction andThen parseJsonAction).async(parse.json) {
-      implicit request: ParsedPreValidateTraderRequest[JsValue] =>
-        preValidateTraderService.submitMessage(request).flatMap {
-          case Right(response) => Future.successful(Ok(Json.toJson(response)))
-          case Left(error)     => Future.successful(error)
-        }
-    }
+  def submit: Action[JsValue] = Action.async(parse.json) { implicit request =>
+    authAction.invokeBlock(
+      request,
+      { authRequest: EnrolmentRequest[JsValue] =>
+        if (appConfig.etdsPreValidateTraderEnabled) {
+          // ETDS request
+          parseJsonAction.refineETDS(authRequest).flatMap {
+            case Right(parsedRequest: ParsedPreValidateTraderETDSRequest[JsValue]) =>
+              preValidateTraderService.submitETDSMessage(parsedRequest).map {
+                case Right(response) =>
+                  response match {
+                    case response200: ExciseTraderValidationETDSResponse           => Ok(Json.toJson(response200))
+                    case response400: PreValidateTraderETDS400ErrorMessageResponse =>
+                      BadRequest(Json.toJson(response400))
+                    case response500: PreValidateTraderETDS500ErrorMessageResponse =>
+                      InternalServerError(Json.toJson(response500))
+                  }
+                case Left(error)     =>
+                  error
+              }
+            case Left(result)                                                      =>
+              Future.successful(result)
+          }
+        } else {
+          // Legacy request
+          parseJsonAction.refine(authRequest).flatMap {
+            case Right(parsedRequest: ParsedPreValidateTraderRequest[JsValue]) =>
+              preValidateTraderService.submitMessage(parsedRequest).map {
```

Comment:
Just for the sake of legibility, you could refactor the core logic for both routes into defs to make it clearer to see that one or the other is legacy vs ETDS by their name rather than needing to comment on it.

Alternatively, the way you've written the case statements to match on type suggests to me that you could merge the refine and refineETDS methods.